app/Models/Concerns/HasPassword.php

```diff
@@ -0,0 +1,160 @@
+<?php
+
+namespace App\Models\Concerns;
+
+use App\Events\User\PasswordChanged;
+use App\Events\User\PasswordRemoved;
+use Carbon\Carbon;
+use Illuminate\Support\Facades\Auth;
+use Illuminate\Support\Facades\Hash;
+use Illuminate\Support\Facades\Session;
+use Laravel\Passport\Token;
+
+trait HasPassword
+{
+    private $passwordAttributeName = 'password';
+    private $passwordRefreshRateAttributeName = 'password_refresh_rate';
+
+    /**
+     * Verify if supplied password is correct for user.
+     *
+     * @param $password
+     * @return bool
+     */
+    public function verifyPassword(string $password): bool
+    {
+        if ($this->{$this->passwordAttributeName} === sha1(sha1($password))) {
```

Comment:
I assume we're holding onto sha1 for legacy reasons? It shouldn't be considered cryptographically secure.

Reply:
Yes, this is straight out of the core code base. I might be able to find a better way to do it in laravel

Reply:
Actually, yes this must be legacy for an old core problem and no longer needed. Removed.